### ISSUE TITLE
NimbleCSV.RFC4180 and trailing CRLF

### ISSUE THREAD

**Issue by wojtekmach** [plataformatec/nimble_csv#17]
For the default parser, trailing `\r` is added:

```elixir
iex> NimbleCSV.RFC4180.parse_string("name,last,year\r\njohn,doe,1986\r\n")
[["john", "doe", "1986\r"]]
```

---

I dealt with it by creating a parser with reversed order of newline markers:

```elixir
iex> NimbleCSV.define(MyCSV, newlines: ["\r\n", "\n"])
iex> MyCSV.parse_string("name,last,year\r\njohn,doe,1986\r\n")
[["john", "doe", "1986"]]
```

but maybe it's worth addressing in RFC4180?

**Comment by josevalim** [comment]
@wojtekmach which version are you using? I thought that was already the case?

https://github.com/plataformatec/nimble_csv/blob/master/lib/nimble_csv.ex#L110

Or maybe the order matters here and that's what is causing issues?

**Comment by wojtekmach** [comment]
I'm using 0.1.1. The order does seem to matter: `newlines: ["\n", "\r\n"]` leaves the trailing `\r`, however `newlines: ["\r\n", "\n"]` does not.

**Comment by josevalim** [comment]
Fixed in master and shipped 0.2.0!

**Comment by wojtekmach** [comment]
@josevalim I re-checked the RFC [1] and it says `CRLF` should be the line separator, so maybe `:line_separator` should be set to `\r\n` in `NimbleCSV.RFC4180`? Sorry I probably could have came up with this sooner 😓 

[1] https://tools.ietf.org/html/rfc4180#section-2
